### PR TITLE
feat: API for creating DID inside wallet

### PR DESCRIPTION
**Pull request by sudeshrshetty**
- added just `DIDCreator` interface inside wallet and provided default
implementation
- actual implementation will be provided in upcoming PR
- related to #283

Signed-off-by: sudesh.shetty <sudesh.shetty@securekey.com>

### CODE REVIEW

File: pkg/wallet/api.go [hunk 1]
```diff
@@ -86,6 +91,19 @@ type Pack interface {
...
+type DIDCreator interface {
+	// Creates new DID document.
+	//
+	//
+	// Returns:
+	//
+	// did: DID document
+	//
+	// error: error
+	NewDID() (*did.Doc, error)
```
Comment (troyronda) [inline pkg/wallet/api.go]: prefer Create prefix rather than New within interfaces.

File: pkg/internal/mock/wallet/mock_wallet.go [hunk 2]
```diff
@@ -45,3 +48,8 @@ func (m *CloseableWallet) PackMessage(envelope *wallet.Envelope) ([]byte, error)
 func (m *CloseableWallet) UnpackMessage(encMessage []byte) (*wallet.Envelope, error) {
 	return nil, nil
 }
+
+// NewDID returns new DID Document
+func (m *CloseableWallet) NewDID() (*did.Doc, error) {
```
Comment (llorllale) [inline pkg/internal/mock/wallet/mock_wallet.go]: We should be able to specify the DID method.

Comment (sudeshrshetty) [reply pkg/internal/mock/wallet/mock_wallet.go]: Will be adding it in upcoming PR under same issue number, added a TODO with issue ID for now.

File: pkg/wallet/api.go (hunk 1 above)
Comment (sudeshrshetty) [reply pkg/wallet/api.go]: done

File: pkg/wallet/api.go [hunk 3]
```diff
@@ -86,6 +91,20 @@ type Pack interface {
 	UnpackMessage(encMessage []byte) (*Envelope, error)
 }
 
+// DIDCreator provide method to create DID document
+type DIDCreator interface {
+	// Creates new DID document.
+	//
+	// TODO :Should be able to specify the DID method as argument (Issue #268).
```
Comment (llorllale) [inline pkg/wallet/api.go]: #268 is a merged pull request

Comment (sudeshrshetty) [reply pkg/wallet/api.go]: Upcoming PR for issue #268, still in progress.

**Pull request merged**